### PR TITLE
docs(feature-targeting): Add README

### LINKED ISSUE #4266: mdc-feature-targeting: Add README and mixin examples
This is a follow-up to #4228, which adds the mdc-feature-targeting package to allow more granular inclusion of styles. That PR doesn't contain a README for the new package, and we should at least explain the top-level API usage from a component authoring perspective.

### PR DESCRIPTION
**Pull request by kfranqueiro**
Fixes #4266.

cc @mmalerba

### CODE REVIEW

File: packages/mdc-feature-targeting/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  @include mdc-feature-targets($feat-structure) {
+    // ...
+  }
+}
+```
+
+Consuming component styles:
+
+```scss
+@import "my-component-mixins";
```
Comment (mmalerba) [inline packages/mdc-feature-targeting/README.md]: nit: `@import "@material/feature-targeting/functions";`

File: packages/mdc-feature-targeting/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+// To include all styles (using the default of mdc-feature-all() defined above):
+@include my-component-core-styles;
+
+// Or, to include a specific subset of styles:
+@include my-component-core-styles(mdc-feature-any(structure));
```
Comment (mmalerba) [inline packages/mdc-feature-targeting/README.md]: This query can be expressed more simply as just `@include my-component-core-styles(structure)`

File: packages/mdc-feature-targeting/README.md (hunk 1 above)
Comment (kfranqueiro) [reply packages/mdc-feature-targeting/README.md]: Whoops. I'd say that's more than just a nit.

File: packages/mdc-feature-targeting/README.md (hunk 2 above)
Comment (kfranqueiro) [reply packages/mdc-feature-targeting/README.md]: I simplified this and added more examples to show more function usage cases.

File: packages/mdc-feature-targeting/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Mixin | Description
+--- | ---
+`mdc-feature-targets($feature-targets...)` | Conditionalizes content to only be emitted if the given feature target(s) is/are queried.
+
+Function | Description
+--- | ---
+`mdc-feature-create-target($feature-query, $targeted-feature)` | Returns a variable to be passed to `mdc-feature-targets` in order to filter emitted styles.
+`mdc-feature-all($feature-queries...)` | Returns a query object which will result in emitting `mdc-feature-targets` blocks that match _all_ of the specified features. Passing no arguments results in all blocks being emitted, which is the most common use case.
+`mdc-feature-any($feature-queries...)` | Returns a query object which will result in emitting `mdc-feature-targets` blocks that match _any_ of the specified features. Passing no arguments results in no blocks being emitted.
```
Comment (abhiomkar) [inline packages/mdc-feature-targeting/README.md]: Question,


`mdc-feature-any(color, typography)` and `mdc-feature-all(color, typography)` are same?

Comment (mmalerba) [reply packages/mdc-feature-targeting/README.md]: Given:

```sass
@mixin test($query: mdc-feature-all()) {
  $feat-color: mdc-feature-create-target($query, color);
  $feat-typography: mdc-feature-create-target($query, typography);

  @include mdc-feature-targets($feat-color) {
    content: 'color';
  }

  @include mdc-feature-targets($feat-typography) {
    content: 'typography';
  }

  @include mdc-feature-targets($feat-color, $feat-typography) {
    content: 'color & typography';
  }
}
```

This:
```sass
@include test(mdc-feature-any(color, typography));
```

Results in:
```css
content: 'color';
content: 'typography';
content: 'color & typography';
```

And this:
```sass
@include test(mdc-feature-all(color, typography));
```

Results in:
```css
content: 'color & typography';
```

**Pull request merged**